readthedocs/proxito/views/serve.py

```diff
@@ -274,6 +286,172 @@ def _get_canonical_redirect_type(self, request):
 
         return None
 
+    def get_using_unresolver(self, request):
+        """
+        Resolve the current request using the new proxito implementation.
+
+        This is basically a copy of the get() method,
+        but adapted to make use of the unresolved to extract the current project, version, and file.
+        """
+        unresolved_domain = request.unresolved_domain
+        # TODO: capture this path in the URL.
+        path = request.path_info
+        try:
+            unresolved = unresolver.unresolve_path(
+                unresolved_domain=unresolved_domain,
+                path=path,
+                append_indexhtml=False,
+            )
+        except (
+            VersionNotFoundError,
+            TranslationNotFoundError,
+            InvalidExternalVersionError,
+        ) as exc:
+            # TODO: find a better way to pass this to the middleware.
+            request.path_project_slug = exc.project.slug
+            raise Http404
+        except UnresolvedPathError as exc:
+            project = exc.project
+            if unresolved_domain.is_from_external_domain:
+                version_slug = unresolved_domain.external_version_slug
+            else:
+                version_slug = None
+
+            # TODO: find a better way to pass this to the middleware.
+            request.path_project_slug = project.slug
+            request.path_version_slug = version_slug
+
+            # Before doing anything else, we need to check for canonical redirects.
+            redirect_type = self._get_canonical_redirect_type(request)
+            if redirect_type:
+                # A canonical redirect can be cached, if we don't have information
+                # about the version, since the final URL will check for authz.
+                if self._is_cache_enabled(project):
+                    self.cache_request = True
+                try:
+                    return self.canonical_redirect(
+                        request=request,
+                        final_project=project,
+                        version_slug=version_slug,
+                        filename=path,
+                        redirect_type=redirect_type,
+                        is_external_version=unresolved_domain.is_from_external_domain,
+                    )
+                except InfiniteRedirectException:
+                    # Don't redirect in this case, since it would break things.
+                    # Disable caching, since we failed to redirect.
+                    self.cache_request = False
+
+            # If the path is not empty, the path doesn't resolve to a proper file.
+            if exc.path not in ["", "/"]:
```

Comment:
I think I can just normalize the path before returning it here, so we would just need to check for `/`. If we want to original value, we have it in the `path` variable.